<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <comment>
------------------------------------------------------------------------------

 Author: Andreas Schenk
 Friedrich Miescher Institute, Basel, Switzerland

 This file is part of CryoFlare

 Copyright (C) 2017-2019 by the CryoFLARE Authors

 This program is free software; you can redistribute it and/or modify it under
 the terms of the GNU General Public License as published by the Free
 Software Foundation; either version 3.0 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
 details.

 You should have received a copy of the GNU General Public License
 along with CryoFlare.  If not, see http://www.gnu.org/licenses/ .

------------------------------------------------------------------------------
</comment>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1599</width>
    <height>1011</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CryoFLARE</string>
  </property>
  <property name="windowIcon">
   <iconset resource="app.qrc">
    <normaloff>:/icons/cryoflare128.png</normaloff>:/icons/cryoflare128.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_11">
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>EPU folder</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="PathEdit" name="avg_source_dir">
        <property name="toolTip">
         <string>EPU project directory</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Movie folder</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="PathEdit" name="stack_source_dir">
        <property name="toolTip">
         <string>Directory where EPU saves raw stacks</string>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="start_stop">
        <property name="toolTip">
         <string>Start/Stop processing</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="app.qrc">
          <normaloff>:/icons/media-playback-start.png</normaloff>
          <normalon>:/icons/media-playback-stop.png</normalon>:/icons/media-playback-start.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="export_report">
        <property name="toolTip">
         <string>Export report</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="app.qrc">
          <normaloff>:/icons/export-plots.png</normaloff>:/icons/export-plots.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="export_data">
        <property name="toolTip">
         <string>Export processed images</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="app.qrc">
          <normaloff>:/icons/document-export-table.png</normaloff>:/icons/document-export-table.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="settings">
        <property name="toolTip">
         <string>Settings</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="app.qrc">
          <normaloff>:/icons/applications-system.png</normaloff>:/icons/applications-system.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tab_widget">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="MicrographsForm" name="micrographs">
       <attribute name="title">
        <string>Micrographs</string>
       </attribute>
      </widget>
      <widget class="GridsquareForm" name="gridsquares">
       <attribute name="title">
        <string>Gridsquares</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1599</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PathEdit</class>
   <extends>QLineEdit</extends>
   <header>pathedit.h</header>
  </customwidget>
  <customwidget>
   <class>GridsquareForm</class>
   <extends>QWidget</extends>
   <header>gridsquareform.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MicrographsForm</class>
   <extends>QWidget</extends>
   <header>micrographsform.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="app.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>start_stop</sender>
   <signal>toggled(bool)</signal>
   <receiver>avg_source_dir</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1345</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start_stop</sender>
   <signal>toggled(bool)</signal>
   <receiver>stack_source_dir</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1345</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>1295</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settings</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1595</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>1330</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start_stop</sender>
   <signal>toggled(bool)</signal>
   <receiver>settings</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1345</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>1595</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_data</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onExport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1445</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_report</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>writeReport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1395</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>1180</x>
     <y>-3</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>avgSourceDirChanged()</signal>
  <signal>stackSourceDirChanged()</signal>
  <slot>onAvgSourceDirBrowse()</slot>
  <slot>onStackSourceDirBrowse()</slot>
  <slot>onDestinationDirBrowse()</slot>
  <slot>onAvgSourceDirTextChanged(QString)</slot>
  <slot>onStackSourceDirTextChanged(QString)</slot>
  <slot>onDestinationDirTextChanged(QString)</slot>
  <slot>onSettings()</slot>
  <slot>onExport()</slot>
  <slot>updateDetails()</slot>
  <slot>updateChart()</slot>
  <slot>selectFromLinearChart(float,float,bool)</slot>
  <slot>selectFromHistogramChart(float,float,bool)</slot>
  <slot>phasePlateBack()</slot>
  <slot>gridSquareBack()</slot>
  <slot>phasePlateClicked(int)</slot>
  <slot>gridSquareClicked(int)</slot>
  <slot>enableSelection(bool)</slot>
  <slot>writeReport()</slot>
 </slots>
</ui>
